isp55e0.c

```diff
@@ -840,10 +840,15 @@ static void load_file(struct device *dev, struct content *info)
 
 	memset(info->buf, 0xff, info->len);
 
-	/* TODO: loop until all read, or use mmap instead. */
-	ret = read(fd, info->buf, statbuf.st_size);
-	if (ret != statbuf.st_size)
-		err(EXIT_FAILURE, "Can't read firmware file");
+	off_t total_read = 0;
+	while (total_read < statbuf.st_size) {
+		int ret = read(fd, info->buf + total_read, statbuf.st_size - total_read);
+		if (ret == 0) break;
```

Comment:
If the code hits this, something is wrong. That loops reads exactly the size of the file. It'd treat that as an error too.

Reply:
Indeed, but in win32 it ends with a read == 0.
i have to check why, as this should mean that the file is not read in full...

Reply:
Ok. never mind. I was stupid. It is solved.